pkg/fuzzer/queue/queue.go

```diff
@@ -0,0 +1,322 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package queue
+
+import (
+	"context"
+	"math/rand"
+	"sync"
+
+	"github.com/google/syzkaller/pkg/ipc"
+	"github.com/google/syzkaller/pkg/signal"
+	"github.com/google/syzkaller/pkg/stats"
+	"github.com/google/syzkaller/prog"
+)
+
+type Request struct {
+	Prog       *prog.Prog
+	NeedSignal SignalType
+	NeedCover  bool
+	NeedHints  bool
```

Comment:
It might. But it seems that we have to address two somehow conflicting needs:
1) In `pkg/fuzzer`, we wants a simpler interface: just specify what kind of coverage/signal we want.
2) In `pkg/vminfo` and `pkg/runtest`, we want full control over the execution parameters.

Wdyt, could it work if we pass into `fuzzer.Config` some "base" execution options that we generate here?
https://github.com/google/syzkaller/blob/fdb4c10ce6d49163d1f57463d7ca0ae22b71e765/syz-manager/rpc.go#L585

Then, the fuzzer will clone the base exec opts, manipulate them directly like here
https://github.com/google/syzkaller/blob/fdb4c10ce6d49163d1f57463d7ca0ae22b71e765/syz-manager/rpc.go#L607-L615

and put it to all `Request` objects.

Reply:
Adding something to fuzzer Config looks reasonable.
I don't understand the second part: if we don't have separate flags, we can't do:
```
 if req.NeedCover { 
 	exec |= ipc.FlagCollectCover 
 } 
```
What are the "base" exec options? Is it sandbox only, or something else?